src/react-decor/react-decor-reflection.ts

```diff
@@ -1,31 +1,30 @@
-import { ComponentType, Component } from 'react';
-import { Wrapper, DecorReactHooks, StatelessElementHook, StatefulElementHook, StatelessDecorReactHooks } from './index';
-import { StateProvider, privateState } from "../core/private-state";
+import { Wrapper } from './index';
+import { privateState } from "../core/private-state";
 import { getGlobalConfig } from "../core/config";
 import { GlobalConfig } from "../core/types";
 
-export function createDecorationReflection<T extends object = any>(id: string) {
+export function reflection<T extends object = any>(id: string) {
     const decorationReflection = privateState<Array<any>, T>(id, (targetObj: T) => []);
 
     return {
-        addHooksToDecorReflection(Comp: T, Wrapped: T, decoratorIdentifier: any): void {
-            const isWrapped = decorationReflection.hasState(Wrapped);
-            const newDecorators = decorationReflection(Wrapped); // new private state will be created if missing
+        registerDecorator(CompBeforeDecoration: T, CompAfterDecoration: T, decoratorIdentifier: any): void {
```

Comment:
arguments names should not indicate react at all:
src and target
orig and dest etc.